lib/client/weblogin.go

```diff
@@ -354,29 +354,34 @@ func SSHAgentSSOLogin(login SSHLogin) (*auth.SSHLoginResponse, error) {
 	// macOS.
 	case teleport.DarwinOS:
 		path, err := exec.LookPath(teleport.OpenBrowserDarwin)
-		if err == nil {
+		if err == nil && tc.Config.Browser != teleport.BrowserNone {
```

Comment:
surround the entire `swich` statement with `if tc.Config.Browser != teleport.BrowserNone` instead, to avoid the unnecessary `$PATH` lookup

Reply:
Ok done.